### PR TITLE
SONARJS-923 Update ParenthesesCheck: raise issue only on double parens

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/ParenthesesCheck.java [hunk 1]
```diff
@@ -19,145 +19,23 @@
...
   @Override
-  public void visitUnaryExpression(UnaryExpressionTree tree) {
-    if (tree.is(Kind.DELETE, Kind.VOID)) {
-      checkExpression(tree.expression());
-    }
-    if (tree.is(Kind.TYPEOF)) {
-      checkTypeOfExpression(tree.expression());
+  public void visitParenthesisedExpression(ParenthesisedExpressionTree tree) {
+    if (tree.expression().is(Kind.PARENTHESISED_EXPRESSION)) {
+      addIssue(tree, MESSAGE);
```
Comment (ivandalbosco) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ParenthesesCheck.java]: @vilchik-elena The RSPEC (rightfully in my opinion) says: primary location on extra "(" and secondary location on extra ")". Your location is the whole expression. This is too much, as in:
```
((function(){
    // all your code here
    var foo = function() {};
    window.onload = foo;
    var a = 1 + ((2 + 3) * 4);
    var b = 2 - (1 + ((2 + 3) * 4));
    var c = (( 1 - 4));
}))();
```
(the whole function is highlighted)

File: javascript-checks/src/test/resources/checks/parentheses.js [hunk 2]
```diff
@@ -1,71 +1,9 @@
 function sayHello() {
-  var a = typeof (37); // Noncompliant {{Remove those useless parentheses.}}
-//               ^^^^
-  var b = typeof 38;
+  var a = typeof ((37)); // Noncompliant {{Remove those useless parentheses.}}
+//               ^^^^^^
```
Comment (ivandalbosco) [inline javascript-checks/src/test/resources/checks/parentheses.js]: @vilchik-elena Suggestion: add a test with triple parentheses. Good for testing precise issue location!

File: javascript-checks/src/main/java/org/sonar/javascript/checks/ParenthesesCheck.java [hunk 3]
```diff
@@ -19,145 +19,23 @@
...
-import org.sonar.plugins.javascript.api.tree.expression.NewExpressionTree;
 import org.sonar.plugins.javascript.api.tree.expression.ParenthesisedExpressionTree;
-import org.sonar.plugins.javascript.api.tree.expression.UnaryExpressionTree;
-import org.sonar.plugins.javascript.api.tree.expression.YieldExpressionTree;
-import org.sonar.plugins.javascript.api.tree.statement.ForObjectStatementTree;
-import org.sonar.plugins.javascript.api.tree.statement.ReturnStatementTree;
-import org.sonar.plugins.javascript.api.tree.statement.ThrowStatementTree;
 import org.sonar.plugins.javascript.api.visitors.DoubleDispatchVisitorCheck;
 
 @Rule(key = "Parentheses")
```
Comment (ivandalbosco) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ParenthesesCheck.java]: @vilchik-elena I could not see the rule in the default profile, although the ticket mentions that it should be added to the default profile.

Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ParenthesesCheck.java]: @ivandalbosco indeed, i want to update default profile later, in one shot

File: javascript-checks/src/main/java/org/sonar/javascript/checks/ParenthesesCheck.java (hunk 1 above)
Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/ParenthesesCheck.java]: @ivandalbosco Indeed, forgot to update it. I will do it

File: javascript-checks/src/test/resources/checks/parentheses.js (hunk 2 above)
Comment (vilchik-elena) [inline javascript-checks/src/test/resources/checks/parentheses.js]: @ivandalbosco done:)

**Pull request merged**